### PR TITLE
guCmpHotFix refactor

### PR DESCRIPTION
**Pull request by sndrs**
Sets up the shared methods for `guCmpHotFix` using `||=` (idea came in my sleep! much cleaner)

```js
export const x = (window.x ||= 'xxx') 
export const y = (window.x ||= 'yyy') 
// `x`, `y` and `window.x` all have the value 'xxx'
```

### CODE REVIEW

File: src/types/window.d.ts [hunk 1]
```diff
@@ -1,24 +1,19 @@
+import { disable } from '../disable';
```
Comment (mxdvl) [inline src/types/window.d.ts:1]: It's currently not used. Is there no rule for unused imports?

```suggestion
```

**Review by mxdvl — APPROVED** [review]
LGTM!

If we’re refactoring, would it be good to add a version check on the `window.guCmpHotFix` to confirm that both instances are using the same version? I’m not sure how that would be achieved, could the build step include setting this as read from `package.json` to the `guCmpHotFix` object?

**Comment by sndrs** [comment]
> If we’re refactoring, would it be good to add a version check on the window.guCmpHotFix to confirm that both instances are using the same version? I’m not sure how that would be achieved, could the build step include setting this as read from package.json to the guCmpHotFix object?

that is a very good point. this is breaking currently, i'll refactor this to maintain the same shape

File: .eslintrc.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
 		"browser": true
 	},
 	"rules": {
+		"no-multi-assign": 0,
```
[the remark this answers is not in the document]
Comment (sndrs) [reply .eslintrc.json:21]: by default, eslint lints against this type of thing:

```js
const x = y = z = 'hi'
x === 'hi' // true
y === 'hi' // true
z === 'hi' // true
```

I'm ambivalent about it usually, but in this case i think it keeps the code much more readable. hence disabling it

**Pull request merged**